File: problem33/p33.py
def digit_cancelling_fractions():
    res = 1
    for i in range(10, 101):
        for j in range(i+1, 101):
            if i % 10 == j // 10 and j % 10 != 0:
                if i / j == (i // 10)/(j % 10):
                    res *= i / j
    print(res)


digit_cancelling_fractions()

File: problem33/p33.c
#include <stdio.h>
#include <stdlib.h>

void digitCancellingFractions(){
    float mult = 1.0;
    for(float i=10.0; i<100; i++){
        for(float j=i+1; j<100; j++){
            if((int)i % 10 == (int)j / 10){
                if(i / j == (float)((int)i/10) / ((int)j % 10)){
                    mult *= i/j;
                }
            }
        }
    }

    printf("%f", mult);
}

int main()
{
    digitCancellingFractions();
    return 0;
}
